UC for portable storage and basic sharing (#149)
w3c/lws-ucs: spec/user-stories.md

## Functional Stories

### <dfn>Generic Portable Storage</dfn>

> Bob wants to store and organize his data in a portable `[=storage=]`, so that data could be accessed from any device, any location, at any time. Bob wants to be free to store any type of resource in the `[=storage=]`. Bob wants resources to be added, updated and/or deleted. Bob also wants the ability and choice to host the `[=storage=]` himself, or delegate the hosting to a `[=storage provider=]`. The `[=storage provider=]` may impose certain limitations, such as of type or size, and Bob may accept that if informed in advance. Bob also wants to ability to move his storage to another `[=storage provider=]`.

Reviewer: "portable" being?

Author: It absolutely is about *protocol* interoperability.
From your comment I understand that you have no issue with the storage being hosted by a provider. If, like you say, portability is implied (and in my mind it definitely is, plenty of prior art examples, like phone numbers), then what happens with identifiers and addresses if Bob decides to change providers. The mechanics of porting the data is not a protocol issue, I agree with that, but the consequences of porting do have a significant impact on the protocol. This was discussed in the WG meetings already.

Reviewer: I think you're reading agreement into my comment that I didn't actually express. I didn't say I have "no issue". I asked you to clarify what you mean by "portable" because the text wasn't clear to me.

When I said portability seems implied, I was pointing out a gap in the text. I wasn't endorsing that the meaning is obvious. So, no need to treat that as confirmation of your view!

Again, I'm asking for clarity on what "portable" covers here. You're shifting to consequences and meeting discussions without addressing the gap in the text. I don't think any reader is expected to know the complete background, what's in your mind, or what was discussed in meetings.

Could you clarity what you've meant in the PR itself as opposed to defending it?

I should add that just because something was discussed in a meeting does not necessarily mean there was group consensus on what to say specifically here. If there was such consensus, please refer to specific lines: https://www.w3.org/2025/05/05-lws-minutes.html . It's perfectly fine if this is your own proposal or preference, but that should be clear and not presented as the group's settled position or whatever.

Reviewer 2: Phone numbers have the (current) advantage of *one* global syntax, starting with the `+`, followed by a `<country code>`, `<area code>`, `local exchange`, and `local specific`. For *decades*, they were *not* portable; the `local exchange` meant that your *always a land-line* telephone was physically located within a few miles of the controlling phone company's local facility.

*Today*, you cannot definitively learn anything from a bare phone number. The <area code> *suggests* that when first issued, the holder of that number was physically located within a predefined area — but once ported or assigned to a cell phone, their physical location was no longer predetermined.

None of this applies to URLs.

Reviewer 2: @hzbarcea — URLs are not inherently tied to a _service provider_; they can just as well be based on a `[user@]host[:port]` [authority](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2) which is based on a domain that is registered to the entity that mints the subsidiary URLs. In such case, everything can be moved from one provider to another without changing a single URL — because dereferencing the authority will automagically result in the new IP addresses rather than the old.

The above is the roughly parallel situation to phone numbers.

The situation where the URLs are based on the service providers `authority` is rather different. Here, dereferencing the URLs after moving the data from one hosting provider to another will result in the old IP addresses, not the new, and web surfers will be served the wrong data.

All told — to provide Linked Web Storage users with the desired level of data portability, all URIs must be based on an `authority` controlled by each data owner, not based on an `authority` controlled by the storage provider.

## Functional Stories

### <dfn>Generic Storage</dfn>

Bob wants to store and organize his data in a `[=storage=]` available online, so that data could be accessed from any device, any location, at any time. Bob wants to be free to store any type of resource in the `[=storage=]`. Bob wants his `[=storage=]` to allow for resources to be added, updated and/or deleted.

### <dfn>Portable Storage</dfn>

Bob wants the ability and choice to host the `[=storage=]` himself, or delegate the hosting to a `[=storage provider=]`. The `[=storage provider=]` may impose certain limitations, such as of type or size, and Bob may accept that if informed in advance. Bob also wants the ability to move (aka port) his `[=storage=]` to another `[=storage provider=]` as desired, without loosing any access to existing data or metadata, so that Bob is not dependent in perpetuity on one `[=storage provider=]`.

Porting a `[=storage=]` from one provider to another should work the same even when Bob hosts the `[=storage=]` himself. In that situation Bob acts as `[=storage provider=]` as well. 

### <dfn>Sharing Access</dfn>

> As an author with authoring permission, Guinan wants to use the sharing interface of her authoring tool to manage access permissions to her article, so that Deanna can review, Seven-of-Nine can co-edit, and Torres can view the progress. All `[=user=]`(s), Guinan, Deanna, Seven-of-Nine and Torres can identify themselves within the authoring tool using their personal online identities.

Reviewer 3: *the authoring tool* might suggest that everyone uses the same tool, is that intended?

As for *identify themselves within the authoring tool*, is this detail important for this use case or maybe that the author can use those online identities when they manage access permissions.

> As an author with authoring permission, Guinan wants to use the sharing interface of her authoring tool to manage access permissions to her article, so that Deanna can review, Seven-of-Nine can co-edit, and Torres can view the progress. All `[=user=]`(s), Guinan, Deanna, Seven-of-Nine and Torres can identify themselves within the authoring tool using their personal online identities.

Reviewer: Fair point. The original wording could've been clearer. It probably should've been improved rather than copied over verbatim from the use case issue, but I'm happy to revise it now.

The phrase might imply everyone is using the same tool, but that's not the intent. The use case assumes only that Guinan manages access using her tool's sharing interface. The others just need to have online identities that the tool can recognise.

```suggestion
As an author with authoring permission, Guinan uses the sharing interface of her authoring tool to manage access to her article, so that Deanna can review, Seven-of-Nine can co-edit, and Torres can view the progress. All `[=user=]`(s), Guinan, Deanna, Seven-of-Nine, and Torres have personal online identities that can be used to assign these permissions.
```

Author: I found the intent pretty clear. All actors use their respective tools. The sharing capability of any and all such tools shall use the protocol in the same way to manage access permissions.

What I would have changed is the name of the actors for consistency, i.e. [Alice and Bob](https://en.wikipedia.org/wiki/Alice_and_Bob). Maybe I'll do that later.

Reviewer: -1 to Alice and Bob.

Standardising on them reinforces a narrow cultural default. They're overused, culturally stale, and signal a lack of inclusivity. And, pretty boring at this point.

Use names from underrepresented ethnic and cultural backgrounds instead. Inclusive naming reflects global diversity, counters bias, and aligns with W3C values (see e.g., [Create inclusive language references](https://github.com/w3c/idcg/issues/17), [Manual of Style](https://www.w3.org/guide/manual-of-style/#inclusive), [Ethical Web Principles](https://www.w3.org/TR/ethical-web-principles/), [Societal Impact Questionnaire](https://w3ctag.github.io/societal-impact-questionnaire/), [Vision for W3C](https://www.w3.org/TR/w3c-vision/).

Resources like [Forebears](https://forebears.io/earth/forenames) and [Google's examples](https://developers.google.com/style/examples#example-person-names) can help to choose representative names.

Reverting to Alice and Bob erases intentional inclusion like using "Guinan" to bring cultural depth. In fact, other examples should be moving away from Alice and Bob.

Reviewer 2: I would suggest avoiding such fictional characters as Guinan, Deanna, Seven-of-Nine, and Torres, because they are similarly culturally tied and carry specific backgrounds based on their fictional arena. I would also suggest using *at least* one "man's" name and one "woman's" name and if possible *at least* one (if not all) ambiguously gendered name.

I think it is also worth everyone familiarizing themselves with W3C's article on [Personal names around the world](https://www.w3.org/International/questions/qa-personal-names), as well as all of the articles linked from the [Human Identity](https://github.com/kdeldycke/awesome-falsehood?tab=readme-ov-file#human-identity)  section of the [Falsehoods Programmers Believe](https://github.com/kdeldycke/awesome-falsehood).

> The sharing interface of the protocl shall ensure that each `[=user=]` receives the appropriate access without confusion. The protocol shall handle updates to access permissions and determine whether changes should trigger notifications to the target `[=user=]`(s).

Reviewer 3: ```suggestion
The sharing interface of the protocol shall ensure that each `[=user=]` receives the appropriate access without confusion. The protocol shall handle updates to access permissions and determine whether changes should trigger notifications to the target `[=user=]`(s).
```
